File: 2502-design-memory-allocator.cpp
class Allocator {
    
public:
    int n;
    int v[1005] = {};
    int mem[1005] = {};
    Allocator(int n) {
        this->n = n;
    }
    
    int allocate(int size, int mID) {
        int free = 0;
        int index;
        for(int i=0; i<n; i++){
            if(free == 0) index = i;
            else if(free >= size) break;
            
            if(v[i] == 0) free++;
            else free = 0;
        }

        if(free>=size){
            for(int i=index; i<index + size; i++){
                v[i] = mID;
                mem[i] = 1;
            }
        }
        else return -1;
        return index;
    }
    
    int free(int mID) {
        int cnt = 0;
        for(int i=0; i<n; i++){
            if(v[i] == mID){
                v[i] = 0;
                cnt++;
            }
        }
        return cnt;
    }
};

// [null, 0, 1, 2, 1, 3, 1, 6, 3, -1, 0]

/**
 * Your Allocator object will be instantiated and called as such:
 * Allocator* obj = new Allocator(n);
 * int param_1 = obj->allocate(size,mID);
 * int param_2 = obj->free(mID);
 */

File: 2502-design-memory-allocator.py
class Allocator(object):

    def __init__(self, n):
        self.n = n
        self.v = [0 for i in range(n)]
        

    def allocate(self, size, mID):
        free = 0
        ind = 0
        for i in range(0, self.n):
            if free == 0: ind = i
            elif (free >= size): break
            
            if(self.v[i] == 0): free+=1
            else: free = 0

        if(free >= size):
            for j in range(ind, ind + size):
                self.v[j] = mID

        return -1 if free < size else ind

    def free(self, mID):
        cnt = 0
        for i in range(0, self.n):
            if (self.v[i] == mID):
                self.v[i] = 0
                cnt += 1
        return cnt
        


# Your Allocator object will be instantiated and called as such:
# obj = Allocator(n)
# param_1 = obj.allocate(size,mID)
# param_2 = obj.free(mID)
